src/bpftrace.cpp

```diff
@@ -409,36 +405,13 @@ void perf_event_printer(void *cb_cookie, void *data, int size __attribute__((unu
   auto args = std::get<1>(bpftrace->printf_args_[printf_id]);
   auto arg_values = bpftrace->get_arg_values(args, arg_data);
 
-  switch (args.size())
-  {
-    case 0:
-      printf(fmt);
-      break;
-    case 1:
-      printf(fmt, arg_values.at(0)->value());
-      break;
-    case 2:
-      printf(fmt, arg_values.at(0)->value(), arg_values.at(1)->value());
-      break;
-    case 3:
-      printf(fmt, arg_values.at(0)->value(), arg_values.at(1)->value(), arg_values.at(2)->value());
-      break;
-    case 4:
-      printf(fmt, arg_values.at(0)->value(), arg_values.at(1)->value(), arg_values.at(2)->value(),
-        arg_values.at(3)->value());
-      break;
-    case 5:
-      printf(fmt, arg_values.at(0)->value(), arg_values.at(1)->value(), arg_values.at(2)->value(),
-        arg_values.at(3)->value(), arg_values.at(4)->value());
-      break;
-    case 6:
-      printf(fmt, arg_values.at(0)->value(), arg_values.at(1)->value(), arg_values.at(2)->value(),
-        arg_values.at(3)->value(), arg_values.at(4)->value(), arg_values.at(5)->value());
-      break;
-    default:
-      std::cerr << "printf() can only take up to 7 arguments (" << args.size() << ") provided" << std::endl;
-      abort();
-  }
+  // Calculate required size by passing NULL
+  int len = format(NULL, 0, fmt, arg_values);
+  char * buffer = new char[len];
```

Comment:
We don't want to be using `new`/`delete` - can this be changed to smart pointers?